pandas/core/array_algos/putmask.py

```diff
@@ -106,8 +106,11 @@ def putmask_smart(values: np.ndarray, mask: np.ndarray, new) -> np.ndarray:
         # preserves dtype if possible
         return _putmask_preserve(values, new, mask)
 
-    # change the dtype if needed
-    dtype, _ = maybe_promote(new.dtype)
+    if values.dtype == bool and new.dtype.kind == "f":
```

Comment:
hmm, shouldn't maybe_promote handle this? (`maybe_promote(new.dtype, values.dtype) ?

Reply:
wouldnt find_common_type make more sense?

Reply:
yes!

Reply:
yah this is nicer, updated